lib/model.d.ts

```diff
@@ -325,7 +325,7 @@ export interface IncludeOptions {
   /**
    * Load further nested related models
    */
-  include?: Includeable[];
+  include?: Includeable[] | IncludeAll[];
```

Comment:
I'm starting to get confused myself, so sorry about that :smile: 
I'll try to summarize it again:

Quoting [this comment](https://github.com/types/sequelize/issues/115#issuecomment-316952963) on the issue I opened, I tried to add a typing that would accept `include: [{ all: true }]`, but no additional properties in that object.

After quite some tests (and our discussions here), I've finally landed on `include?: Includeable[] | { all: true }[];` being the only option to do so, where `Includeable` stays unmodifed as `type Includeable = Model | Association | IncludeOptions;`

To verify the typings were working correctly, I added two "testcases", one that would succeed, and one that should fail:

```TypeScript
// should succeed, only { all: true } provided
MyModel.findAll({
  include: [{
    all: true
  }]
});

// should fail, additional properties beside { all: true } have been added
// since this is not allow, tsc should raise an error
MyModel.findAll({
  include: [{
    all: true,
    include: [{
      model: AssociatedModel,
      where: { state: Sequelize.col('project.state') },
      limit: 1,
      separate: true,
      order: [['id', 'DESC']]
    }]
  }]
});
```

The second test obviously has to be removed before committing the changes, but I've just included it for now in order to verify your requirement gets fulfilled.

Hope this clears it up a bit. Happy to hop onto Slack or likewise as well if that helps.